packages/massa-web3/src/experimental/smartContract.ts

```diff
@@ -157,7 +163,9 @@ export class SmartContract {
     client: BlockchainClient,
     contractAddress: string
   ): SmartContract {
-    return new SmartContract(client, contractAddress)
+    // Used to validate the address
+    const address = Address.fromString(contractAddress)
+    return new SmartContract(client, address.toString())
```

Comment:
why not take a Address type in contractAddress parameter instead of a string ?

Reply:
I agree with Pierre on this. As a rule of thumb, you should take an address. What use case do you have in mind to justify the string parameter?